src/middlewares/error_handlers.cr

```diff
@@ -1,16 +1,74 @@
+require "http"
+require "json"
+
+struct ErrorData
+  include JSON::Serializable
+  property type : String
+  property message : String?
+  property status_code : HTTP::Status?
+
+  def initialize(@type, @message, @status_code = nil)
+  end
+
+  def self.from(error : Exception)
+    if error.responds_to? :status_code
```

Comment:
I think it's possible to avoid this, by adding a method overload, more restrictive than `Exception`

Reply:
updated

Reply:
I wish that worked... :slightly_frowning_face: